api/server/router/grpc/grpc_routes.go

```diff
@@ -0,0 +1,50 @@
+package grpc // import "github.com/docker/docker/api/server/router/grpc"
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"golang.org/x/net/http2"
+)
+
+func (gr *grpcRouter) serveGRPC(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
+	if r.ProtoMajor == 2 {
+		gr.grpcServer.ServeHTTP(w, r)
```

Comment:
how does a client handle this different behaviour?

Reply:
Apparently Docker doesn't support http2 by default, so i'll remove this.